Release instructions template (#488)
WAM2layers/WAM2layers: docs/develop.md

These instructions are intended for core developers.

> - Create a new issue using the issue template `new_release`
> - Follow the steps in this issue template
> - After the release close the associated issue

Reviewer: I've added the "close the issue" step to the template. This doesn't really need to be a bullet list anymore, right? Also we might insert a direct link.

```suggestion
To make a new release, create an issue using the issue template [`New release`](https://github.com/WAM2layers/WAM2layers/issues/new?template=new-release.md) and follow the steps outlined in this issue.
```
